### ISSUE TITLE
cranelift_preopt panic

### ISSUE THREAD

**Issue by bjorn3** [CraneStation/cranelift#611]
```rust
function u0:0(i64, i64) -> i64 fast {
    ss0 = explicit_slot 16
    ss1 = explicit_slot 16

ebb0(v0: i64, v1: i64):
    v3 -> v0
    v4 -> v1
    nop 
    v2 = stack_addr.i64 ss0
    jump ebb1

ebb1:
    nop 
    v5 = isub.i64 v3, v4
    v6 = iconst.i8 0
    v7 = stack_addr.i64 ss1
    store v5, v7
    store v6, v7+8
    v8 = load.i64 v7
    store v8, v2
    v9 = load.i64 v7+8
    store v9, v2+8
    v10 = iadd_imm.i64 v2, 8
    v11 = load.i8 v10
    v12 = uextend.i32 v11
    brz v12, ebb2
    trap user0

ebb2:
    nop 
    v13 = load.i64 v2
    return v13
}
```

```
thread 'main' panicked at 'Value is not an instruction result', /Users/bjorn/Documents/cranelift/lib/codegen/src/ir/dfg.rs:356:18
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::_print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: rustc::util::common::panic_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::begin_panic
   7: cranelift_codegen::ir::dfg::ValueDef::unwrap_inst
   8: cranelift_preopt::constant_folding::resolve_value_to_imm
   9: cranelift_preopt::constant_folding::fold_constants
  10: cranelift_preopt::optimize
  11: rustc_codegen_cranelift::base::trans_fn
  12: rustc_codegen_cranelift::base::trans_mono_item
  13: rustc_codegen_cranelift::codegen_mono_items::{{closure}}
  14: rustc_codegen_cranelift::unimpl::try_unimpl::{{closure}}
  15: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  16: std::panicking::try::do_call
  17: __rust_maybe_catch_panic
  18: std::panicking::try
  19: std::panic::catch_unwind
  20: rustc_codegen_cranelift::unimpl::try_unimpl
  21: rustc_codegen_cranelift::codegen_mono_items
  22: <rustc_codegen_cranelift::CraneliftCodegenBackend as rustc_codegen_utils::codegen_backend::CodegenBackend>::codegen_crate
  23: rustc::util::common::time
  24: rustc_driver::driver::phase_4_codegen
  25: rustc_driver::driver::compile_input::{{closure}}
  26: rustc::ty::context::tls::enter_context
  27: <std::thread::local::LocalKey<T>>::with
  28: rustc::ty::context::TyCtxt::create_and_enter
  29: rustc_driver::driver::compile_input
  30: rustc_driver::run_compiler_with_pool
  31: <scoped_tls::ScopedKey<T>>::set
  32: rustc_driver::run_compiler
  33: syntax::with_globals
  34: __rust_maybe_catch_panic
  35: rustc_driver::run
  36: rustc_driver::main
  37: std::rt::lang_start::{{closure}}
  38: std::panicking::try::do_call
  39: __rust_maybe_catch_panic
  40: std::rt::lang_start_internal
  41: main
```

**Comment by sunfishcode** [comment]
@lachlansneff Could you take a look?

**Comment by lachlansneff** [comment]
Sure, I'll have time this evening.

**Comment by lachlansneff** [comment]
Fixed in #620.
